airflow/jobs/triggerer_job_runner.py

```diff
@@ -538,10 +539,13 @@ async def cancel_triggers(self):
         This allows the cleanup job to delete them.
         """
         while self.to_cancel:
-            trigger_id = self.to_cancel.popleft()
+            trigger_id, reason = self.to_cancel.popleft()
             if trigger_id in self.triggers:
+                # Add more context here before cancelling
+                to_be_cancelled = self.triggers[trigger_id]
+                to_be_cancelled["termination_reason"] = reason
                 # We only delete if it did not exit already
-                self.triggers[trigger_id]["task"].cancel()
+                to_be_cancelled["task"].cancel()
```

Comment:
I couldn’t find where this `to_be_cancelled` is used. Why do we store things into this dict that’s never read by anyone?

Reply:
Ah, it is the same object in `self.triggers` so the additional information is stored there. Nevermind.